Src/Model/StrategyModel.php

```diff
@@ -0,0 +1,83 @@
+<?php
+namespace Aot\Model;
+
+class StrategyModel
+{
+    private $format = [];
+    private $method = [];
+
+    public function __construct()
+    {
+        $this->format = ['csv','json'];
+        $this->method = ['show'];
+    }
+
+    /**
+     * check $format and $method and lead to the function choseFormat
+     * @param $format
+     * @param $method
+     * @param null $id
+     * @return array
+     * @throws \Exception
+     */
+    public function apiService($format, $method, $id = null)
+    {
+        $this->checkFormat($format);
+        $this->checkMethod($method);
+        if ($id) {
+            return $this->choseFormat($format, $method, $id);
+        }
+        return $this->choseFormat($format, $method);
+    }
+
+    /**
+     * check $format
+     * @param $format
+     * @throws \Exception
+     */
+    private function checkFormat($format)
+    {
+        if (!in_array($format, $this->format))
+        {
+            throw new \Exception("format invalid");
+        }
+    }
+
+    /**
+     * check $method
+     * @param $method
+     * @throws \Exception
+     */
+    private function checkMethod($method)
+    {
+        if (!in_array($method, $this->method))
+        {
+            throw new \Exception("method invalid");
+        }
+    }
+
+    /**
+     * define the method and lead to the good class
+     * @param $format
+     * @param $method
+     * @param null $id
+     * @return array
+     * @throws \Exception
+     */
+    private function choseFormat($format, $method, $id = null)
```

Comment:
The code dealing with format represents your factory and this is excellent!
However, (except it has to be a class), the code is duplicated. If the name of `jsonMethod` and `csvMethod` are the same, you can define an `$adapter` at the beginning of the method and play the code only once